pkg/ioutil/ioutil.go

```diff
@@ -212,6 +212,7 @@ func CopyAligned(w *os.File, r io.Reader, alignedBuf []byte, totalSize int64) (i
 			if err = disk.DisableDirectIO(w); err != nil {
 				return remainingWritten, err
 			}
+			defer w.Sync()
```

Comment:
fsync is per file. Even if you are still doing it for the last block it will still sync the entire file.

fsync is a false sense of security that either tanks performance or has no effect whatsoever because some part of the chain ignores it.

We had this discussion less than half a year ago with someone else trying to add it.

If power dies you loose data anyway and we have strong detection of broken content. So unless you have a really good reason to add it, please don't.

Reply:
Yes but we write with O_DIRECT until that point so it won't really block the entire file, it is just the last blocks that which were written O_DIRECT need to be flushed. 

The idea behind this change is that O_DIRECT would have persisted everything to disk until the point of last remaining blocks, which need to be fsynced so to gain the same O_DIRECT behavior of sort.

The assumption in this change is that O_DIRECT once disabled we perform a sync of the remaining writes.

Reply:
Now unless that's not how it works it again flushes everything to disk then yes, it is not going to be a great change since the benefit, in general, is small.